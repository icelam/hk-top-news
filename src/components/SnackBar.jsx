import React from 'react';
import PropTypes from 'prop-types';
import { makeStyles } from '@material-ui/core/styles';

/* Material UI Components */
import Button from '@material-ui/core/Button';
import SnackbarContent from '@material-ui/core/SnackbarContent';

const useStyles = makeStyles({
  root: {
    position: 'fixed',
    bottom: 0,
    left: 0,
    width: '100%',
    'box-sizing': 'border-box'
  },
  snackbar: {
    borderRadius: 0
  }
});

const SnackBar = ({ message, buttonText, buttonAction }) => {
  const classes = useStyles();
  return (
    <div className={classes.root}>
      <SnackbarContent
        open
        className={classes.snackbar}
        message={message}
        action={
          (
            <Button color="secondary" onClick={buttonAction}>
              {buttonText}
            </Button>
          )
        }
      />
    </div>
  );
};

SnackBar.propTypes = {
  message: PropTypes.string.isRequired,
  buttonText: PropTypes.string.isRequired,
  buttonAction: PropTypes.func.isRequired
};

export default SnackBar;
