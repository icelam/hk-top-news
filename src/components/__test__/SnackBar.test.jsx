import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import SnackBar from '../SnackBar';

const message = 'Test Message';
const buttonText = 'OK';
const buttonAction = jest.fn();

describe('Snack Bar', () => {
  it('renders correctly', () => {
    const { asFragment } = render(
      <SnackBar message={message} buttonText={buttonText} buttonAction={buttonAction} />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render with correct message', () => {
    const { container } = render(
      <SnackBar message={message} buttonText={buttonText} buttonAction={buttonAction} />
    );
    expect(container.querySelector('.MuiSnackbarContent-message')).toHaveTextContent(message);
  });

  it('should render button with correct label', () => {
    const { container } = render(
      <SnackBar message={message} buttonText={buttonText} buttonAction={buttonAction} />
    );
    expect(container.querySelector('button')).toBeInTheDocument();
    expect(container.querySelector('.MuiButton-label')).toHaveTextContent(buttonText);
  });

  it('should trigger buttonAction when clicking on button', () => {
    const { container } = render(
      <SnackBar message={message} buttonText={buttonText} buttonAction={buttonAction} />
    );
    const button = container.querySelector('button');
    fireEvent.click(button);
    expect(buttonAction).toHaveBeenCalledTimes(1);
  });
});
