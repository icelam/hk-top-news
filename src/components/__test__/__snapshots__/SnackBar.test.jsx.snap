// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snack Bar renders correctly 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-root-1"
  >
    <div
      class="MuiPaper-root MuiSnackbarContent-root makeStyles-snackbar-2 MuiPaper-elevation6"
      open=""
      role="alert"
    >
      <div
        class="MuiSnackbarContent-message"
      >
        Test Message
      </div>
      <div
        class="MuiSnackbarContent-action"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSecondary"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            OK
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
